### PR TITLE
New Pull Request

### PR DESCRIPTION
**Pull request by Chenxiao-Tian**

### CODE REVIEW

File: ci.yml
Comment (jyl17) [inline ci.yml:1]: It'll be good to combine the file with the existing one in .github/workflows

**Review by Chenxiao-Tian — COMMENTED** [review]
CI Testing

File: test_sampling.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import pytest
```
Comment (jyl17) [inline test_sampling.py]:
```suggestion
```

File: test_sampling.py
Comment (jyl17) [inline test_sampling.py:1]: Can be moved to the tests folder

File: tests/ci.yml
Comment (jyl17) [inline tests/ci.yml:1]: Should be moved to .github/workflows,  combining with the existing ci.yml in that folder

File: tests/test_sampling.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import pytest
```
Comment (jyl17) [inline tests/test_sampling.py]:
```suggestion
```

File: tests/test_sampling.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+from your_script_name import direct_sampling_with_times  # Replace with your actual script name
```
Comment (jyl17) [inline tests/test_sampling.py:1]: Might need some packaging to get rid of this

File: git
Comment (jyl17) [inline git:1]: Can be removed

**Review by jyl17 — CHANGES_REQUESTED** [review]
I think the commits should be merged into main (not other branches)

File: tests/ci.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.8'  # Or any version you need
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install numpy pytest  # Add any other dependencies
+    - name: Test with pytest
+      run: |
+        pytest
```
Comment (jyl17) [inline tests/ci.yml]:
```suggestion
```

File: tests/ci.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.8'  # Or any version you need
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install numpy pytest  # Add any other dependencies
+    - name: Test with pytest
+      run: |
```
Comment (jyl17) [inline tests/ci.yml]:
```suggestion
      run: python -m pytest
```

File: tests/ci.yml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.8'  # Or any version you need
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install numpy pytest  # Add any other dependencies
```
Comment (jyl17) [inline tests/ci.yml]:
```suggestion
        pip install -e .[test]  # test is defined in pyproject.toml and contain all dependencies
```

**Review by jyl17 — COMMENTED** [review]
might be good to merge this ci.yml file with the existing one

File: tests/test_sampling.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
```
Comment (jyl17) [inline tests/test_sampling.py:1]:
```suggestion
import ECMC
```

**Review by jyl17 — COMMENTED** [review]
need package for test to pass successful

File: tests/ci.yml [hunk 8]
```diff
@@ -0,0 +1,21 @@
+name: Python CI
+
+on: [push, pull_request]
```
Comment (Xuyang-Xu) [inline tests/ci.yml:3]: should be changed to
on:
  pull_request:
  push:
    branches:
      - main

**Pull request merged**